### PR TITLE
WFCORE-6975 Add convenience method to combine a ServiceDependency with another.

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFCORE-6975

Based on https://github.com/wildfly/wildfly-core/pull/6156 to avoid conflicts.

### CODE REVIEW

File: service/src/main/java/org/wildfly/service/Dependency.java [hunk 1]
```diff
@@ -18,6 +19,28 @@
  */
 public interface Dependency<B extends ServiceBuilder<?>, V> extends Consumer<B>, Supplier<V> {
 
+    @Override
+    default Dependency<B, V> andThen(Consumer<? super B> after) {
+        return new Dependency<>() {
```
Comment (yersan) [inline service/src/main/java/org/wildfly/service/Dependency.java:24]: Should not do a sanity check `Objects.requireNonNull(after);`?

File: service/src/main/java/org/wildfly/service/ServiceDependency.java [hunk 2]
```diff
@@ -15,6 +18,22 @@
  */
 public interface ServiceDependency<V> extends Dependency<ServiceBuilder<?>, V> {
 
+    @Override
+    default ServiceDependency<V> andThen(Consumer<? super ServiceBuilder<?>> after) {
+        return new ServiceDependency<>() {
```
Comment (yersan) [inline service/src/main/java/org/wildfly/service/ServiceDependency.java:23]: Missed Objects.requireNonNull(after); sanity check?

File: subsystem/src/main/java/org/wildfly/subsystem/service/ServiceDependency.java [hunk 3]
```diff
@@ -25,6 +27,22 @@
  */
 public interface ServiceDependency<V> extends Dependency<RequirementServiceBuilder<?>, V> {
 
+    @Override
+    default ServiceDependency<V> andThen(Consumer<? super RequirementServiceBuilder<?>> after) {
+        return new ServiceDependency<>() {
```
Comment (yersan) [inline subsystem/src/main/java/org/wildfly/subsystem/service/ServiceDependency.java:32]: Missed Objects.requireNonNull(after); sanity check?

File: service/src/main/java/org/wildfly/service/Dependency.java (hunk 1 above)
Comment (pferraro) [reply service/src/main/java/org/wildfly/service/Dependency.java:24]: Sounds good.

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Thanks @pferraro

**Pull request merged**